### ISSUE TITLE
Internal Server Error when visiting localhost:6300

### ISSUE THREAD

**Issue by alloygoh** [CERT-Polska/drakvuf-sandbox#100]
**Describe the bug**

After completing the necessary steps to complete the installation of the sandbox, I tried to go to http://localhost:6300. Instead of the the web UI, I was greeted with an internal server error. 

**How to reproduce**

1. Install sandbox as instructed
2. Execute `sudo draksetup postinstall`
3. Visit localhost:6300

**Output of the [status checking commands](https://github.com/CERT-Polska/drakvuf-sandbox#checking-service-status)**

### systemctl status drak-web.service
```
drak-web.service - drak-web service
   Loaded: loaded (/etc/systemd/system/drak-web.service; enabled; vendor preset: enabled)
   Active: active (running) since Sat 2020-05-09 23:49:53 PDT; 10min ago
 Main PID: 783 (uwsgi)
    Tasks: 6 (limit: 2207)
   CGroup: /system.slice/drak-web.service
           ├─ 783 /opt/venvs/drakcore/bin/uwsgi --ini /etc/drakcore/uwsgi.ini
           ├─1014 /opt/venvs/drakcore/bin/uwsgi --ini /etc/drakcore/uwsgi.ini
           ├─1016 /opt/venvs/drakcore/bin/uwsgi --ini /etc/drakcore/uwsgi.ini
           ├─1018 /opt/venvs/drakcore/bin/uwsgi --ini /etc/drakcore/uwsgi.ini
           ├─1029 /opt/venvs/drakcore/bin/uwsgi --ini /etc/drakcore/uwsgi.ini
           └─1030 /opt/venvs/drakcore/bin/uwsgi --ini /etc/drakcore/uwsgi.ini

May 09 23:49:59 ubuntu uwsgi[783]: *** WARNING: you are running uWSGI as root !!! (use the --uid flag) ***
May 09 23:49:59 ubuntu uwsgi[783]: *** uWSGI is running in multiple interpreter mode ***
May 09 23:49:59 ubuntu uwsgi[783]: spawned uWSGI master process (pid: 783)
May 09 23:49:59 ubuntu uwsgi[783]: spawned uWSGI worker 1 (pid: 1014, cores: 1)
May 09 23:49:59 ubuntu uwsgi[783]: spawned uWSGI worker 2 (pid: 1016, cores: 1)
May 09 23:50:00 ubuntu uwsgi[783]: spawned uWSGI worker 3 (pid: 1018, cores: 1)
May 09 23:50:00 ubuntu uwsgi[783]: spawned uWSGI worker 4 (pid: 1029, cores: 1)
May 09 23:50:00 ubuntu uwsgi[783]: spawned uWSGI http 1 (pid: 1030)
May 09 23:52:48 ubuntu uwsgi[783]: --- no python application found, check your startup logs for errors ---
May 09 23:52:48 ubuntu uwsgi[783]: [pid: 1014|app: -1|req: -1/1] 127.0.0.1 () {36 vars in 563 bytes} [Sat May  9 23:52:48 2020] GET / => generated 21 bytes in 2 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 0)
```

### sudo systemctl status drakrun@1.service
```
drakrun@1.service - drakrun service
   Loaded: loaded (/etc/systemd/system/drakrun@.service; indirect; vendor preset: enabled)
   Active: active (running) since Sat 2020-05-09 23:50:07 PDT; 16min ago
 Main PID: 1334 (drakrun)
    Tasks: 1 (limit: 2207)
   CGroup: /system.slice/system-drakrun.slice/drakrun@1.service
           └─1334 /opt/venvs/drakrun/bin/python /opt/venvs/drakrun/bin/drakrun 1

May 09 23:50:07 ubuntu systemd[1]: Started drakrun service.
May 09 23:52:07 ubuntu drakrun[1334]: iptables: Bad rule (does a matching rule exist in that chain?).
May 09 23:52:07 ubuntu drakrun[1334]: iptables: Bad rule (does a matching rule exist in that chain?).
May 09 23:52:07 ubuntu drakrun[1334]: [2020-05-09 23:52:07,174][INFO] Service karton.drakrun-prod started
May 09 23:52:07 ubuntu drakrun[1334]: [2020-05-09 23:52:07,187][INFO] Service binds created.
May 09 23:52:07 ubuntu drakrun[1334]: [2020-05-09 23:52:07,195][INFO] Binding on: {'type': 'sample', 'stage': 'recognized', 'platform': 'win32'}
May 09 23:52:07 ubuntu drakrun[1334]: [2020-05-09 23:52:07,198][INFO] Binding on: {'type': 'sample', 'stage': 'recognized', 'platform': 'win64'}
```

### tree /var/lib/drakcore
```
.
└── minio
    ├── drakrun
    ├── karton2
    └── .minio.sys
        ├── backend-encrypted
        ├── buckets
        │   ├── .bloomcycle.bin
        │   ├── .minio.sys
        │   │   └── buckets
        │   │       ├── .bloomcycle.bin
        │   │       │   └── fs.json
        │   │       ├── .usage-cache.bin
        │   │       │   └── fs.json
        │   │       └── .usage.json
        │   │           └── fs.json
        │   ├── .tracker.bin
        │   ├── .usage-cache.bin
        │   └── .usage.json
        ├── config
        │   ├── config.json
        │   └── iam
        │       └── format.json
        ├── format.json
        ├── multipart
        └── tmp
            ├── 64f8397f-8834-4223-83c1-336df8a504ef
            ├── 72d1d724-222f-44a6-810c-a24e2c742133
            ├── b7219802-97ef-4e4a-b647-55979848954f
            └── f8c8d0cb-6d09-4a91-bc9b-53fbebf64459

18 directories, 11 files
```

**Comment by icedevml** [comment]
Please try to stop the `drak-web` service:
```
systemctl stop drak-web
```

and run it just manually in the foreground:
```
/opt/venvs/drakcore/bin/uwsgi --ini /etc/drakcore/uwsgi.ini
```

Could you post the full log from the above command?

**Comment by icedevml** [comment]
Also, please verify if you are using one of the supported distributions: Ubuntu 20.04 Focal, Ubuntu 18.04 Bionic or Debian Buster. There are separate Debian packages for each distribution, so the proper one has to be downloaded and installed.

Could you also tell what is your distribution that you are trying right now?

**Comment by alloygoh** [comment]
> Please try to stop the `drak-web` service:
> 
> ```
> systemctl stop drak-web
> ```
> 
> and run it just manually in the foreground:
> 
> ```
> /opt/venvs/drakcore/bin/uwsgi --ini /etc/drakcore/uwsgi.ini
> ```
> 
> Could you post the full log from the above command?

### /opt/venvs/drakcore/bin/uwsgi --ini /etc/drakcore/uwsgi.ini

```
[uWSGI] getting INI configuration from /etc/drakcore/uwsgi.ini
*** Starting uWSGI 2.0.17.1 (64bit) on [Sun May 10 00:09:42 2020] ***
compiled with version: 7.5.0 on 23 April 2020 03:00:00
os: Linux-5.3.0-51-generic #44~18.04.2-Ubuntu SMP Thu Apr 23 14:27:18 UTC 2020
nodename: ubuntu
machine: x86_64
clock source: unix
detected number of CPU cores: 2
current working directory: /home/sandbox
detected binary path: /opt/venvs/drakcore/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
chdir() to /var/lib/drakcore
your processes number limit is 7358
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uWSGI http bound on 0.0.0.0:6300 fd 4
uwsgi socket 0 bound to TCP address 127.0.0.1:44521 (port auto-assigned) fd 3
Python version: 3.6.9 (default, Apr 18 2020, 01:56:04)  [GCC 8.4.0]
Python main interpreter initialized at 0x556d306c7e60
python threads support enabled
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 364520 bytes (355 KB) for 4 cores
*** Operational MODE: preforking ***
Traceback (most recent call last):
  File "/opt/venvs/drakcore/lib/python3.6/site-packages/drakcore/app.py", line 16, in <module>
    from drakcore.pstree import generate_process_tree
  File "/opt/venvs/drakcore/lib/python3.6/site-packages/drakcore/pstree.py", line 2, in <module>
    from dataclasses import dataclass, field
ModuleNotFoundError: No module named 'dataclasses'
unable to load app 0 (mountpoint='') (callable not found or import error)
*** no app loaded. going in full dynamic mode ***
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 16627)
spawned uWSGI worker 1 (pid: 16631, cores: 1)
spawned uWSGI worker 2 (pid: 16632, cores: 1)
spawned uWSGI worker 3 (pid: 16633, cores: 1)
spawned uWSGI worker 4 (pid: 16634, cores: 1)
spawned uWSGI http 1 (pid: 16635)
```

**Comment by alloygoh** [comment]
> Also, please verify if you are using one of the supported distributions: Ubuntu 20.04 Focal, Ubuntu 18.04 Bionic or Debian Buster. There are separate Debian packages for each distribution, so the proper one has to be downloaded and installed.
> 
> Could you also tell what is your distribution that you are trying right now?

Ubuntu 18.04.2 Kernel 5.3.0-51-generic

**Comment by icedevml** [comment]
Seems like you have Python 3.6 set as default interpreter, which is strange because normally Ubuntu 18.04 is shipped with Python 3.7. We will inspect this issue further. In the meantime you can work this around using:

```
/opt/venvs/drakcore/bin/pip install dataclasses
```

then please restart the `drak-web` service.

```
systemctl restart drak-web
```

**Comment by alloygoh** [comment]
ok it works now thanks!!
